redisdb/datadog_checks/redisdb/redisdb.py

```diff
@@ -238,31 +238,38 @@ def _check_key_lengths(self, conn, instance, tags):
         Compute the length of the configured keys across all the databases
         """
         key_list = instance.get('keys')
+        instance_db = instance.get('db')
 
-        if key_list is None:
-            return
-
-        if not isinstance(key_list, list) or len(key_list) == 0:
+        if not isinstance(key_list, list) or not key_list:
             self.warning("keys in redis configuration is either not a list or empty")
             return
 
+        warn_on_missing_keys = is_affirmative(instance.get("warn_on_missing_keys", True))
+
         # get all the available databases
         databases = list(conn.info('keyspace'))
         if not databases:
             self.warning("Redis database is empty")
+            if warn_on_missing_keys:
+                for key in key_list:
+                    key_tags = ['key:{}'.format(key)]
+                    if instance_db:
+                        key_tags.append('redis_db:db{}'.format(instance_db))
+                    key_tags.extend(tags)
+                    self.gauge('redis.key.length', 0, tags=key_tags)
```

Comment:
we are submitting same metric with different tags? is that ok?  what about submitting it just once with all the missing tags?

Reply:
I don't know what's the behavior when you have several value for the same tag.